### ISSUE TITLE
Is basePath a supported property of <ArrayField>?

### ISSUE THREAD

**Issue by Jeff17Robbins** [marmelab/react-admin#5192]
[This [started out on StackOverflow](https://stackoverflow.com/questions/58759554/custom-resource-for-chip-inside-singlefield-react-admin/63603324#63603324), but needs clarification here.]

Is `basePath` a supported prop of `<ArrayField>`?  

I'm hoping the answer is "yes", and I simply missed it in the documentation.  Or, perhaps the documentation is missing this?

**Comment by djhi** [comment]
It is, you can check the source code.

**Comment by Jeff17Robbins** [comment]
If it is missing from the documentation, isn't that an issue? How does the documentation get improved if you don't track issues with it?

**Comment by fzaninotto** [comment]
`basePath` is kind of an internal API. You haven't explained your use case, so I don't understand why we should make it public.

**Comment by Jeff17Robbins** [comment]
I apologize.  I shared the SO link above since another user also had the same problem but should have made this issue self-contained. 

The use case involves the `<List>` View of ResourceA, which has a field which is a list of ResourceB.  

Inside ResourceA's `<List>`` View, I tried to do this

```
<ArrayField  source="ResourceB">
  <SingleFieldList>
    <ChipField source="name" />
  </SingleFieldList>
</ArrayField>
```

but I also wanted the automatic "click" behavior on the `<ChipField>` to take me to the `<Edit>` View of that particular ResourceB item.  As is, the automatically generated URL on the `<ChipField>` is to ResourceA followed by the ResourceB serialization, which throws an error when you click on the `<ChipField>`.

I then tried this

```
<ArrayField source="ResourceB" fieldKey="id">
  <SingleFieldList>
    <ChipField source="name" />
  </SingleFieldList>
</ArrayField>
```

Adding the documented `fieldKey` reduces the stuff appended to the ResourceA url, but still doesn't work because `id` is an id for ResourceB, not for ResourceA.

Only by adding the undocumented `basePath` did I get the behavior I was looking for

```
<ArrayField basePath="/ResourceB" source="ResourceB" fieldKey="id">
  <SingleFieldList>
    <ChipField source="name" />
  </SingleFieldList>
</ArrayField>
```

I'm new at react-admin (which I really really like!) and so I may very well be missing a better way to do this.  But, since I saw an unanswered question on the same exact problem on Stack Overflow, it made me think that maybe this was a documentation issue, since it is hard to tell if a property is missing by accident, or on purpose!
